libraries/cms/application/cms.php

```diff
@@ -114,103 +114,15 @@ public function __construct(JInput $input = null, Registry $config = null, JAppl
 		}
 
 		// Enable sessions by default.
-		if (is_null($this->config->get('session')))
+		if (is_null($this->getContainer()->get('config')->get('session')))
```

Comment:
Should not the DI container being part of the application instead of the application part of the container. What is the benefit of having the application inside the container? Accessing the container would then also be problematic in the component itself. The index.php (Frontcontroller) should be as small as possible.